pkg/builder/action_merge_meta_data_builder.go

```diff
@@ -187,3 +187,120 @@ func (rb *ResizeMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error)
 	mergeSpec.ExecutionTargets = executionTargetList
 	return mergeSpec, nil
 }
+
+
+
+
+// Horizontal Scale Merge Policy DTO builder
+type HorizontalScaleMergePolicyBuilder struct {
+	entityType                *proto.EntityDTO_EntityType
+	aggregationTargets        []*ActionAggregationTargetBuilder
+	chainedAggregationTargets []*ActionDeDuplicateAndAggregationTargetBuilder
+	commTypes                 []*CommodityMergeData
+	entityFilters              []*proto.EntityDTO_WorkloadControllerData
+}
+
+func NewHorizontalScaleMergeSpecBuilder() *HorizontalScaleMergePolicyBuilder {
+	return &HorizontalScaleMergePolicyBuilder{}
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) ForEntityType(entityType proto.EntityDTO_EntityType) *HorizontalScaleMergePolicyBuilder {
+	hsb.entityType = &entityType
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) AggregateBy(mergeTarget *ActionAggregationTargetBuilder) *HorizontalScaleMergePolicyBuilder {
+	hsb.aggregationTargets = append(hsb.aggregationTargets, mergeTarget)
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) DeDuplicateAndAggregateBy(mergeTarget *ActionDeDuplicateAndAggregationTargetBuilder) *HorizontalScaleMergePolicyBuilder {
+	hsb.chainedAggregationTargets = append(hsb.chainedAggregationTargets, mergeTarget)
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) ForWorkloadControllerDataFilter(controllerData proto.EntityDTO_WorkloadControllerData) *HorizontalScaleMergePolicyBuilder  {
+	hsb.entityFilters = append(hsb.entityFilters , &controllerData)
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) ForCommodity(commType proto.CommodityDTO_CommodityType) *HorizontalScaleMergePolicyBuilder  {
+	comm := &CommodityMergeData{
+		commType: commType,
+	}
+	hsb.commTypes = append(hsb.commTypes, comm)
+	return hsb
+}
+
+func (hsb *HorizontalScaleMergePolicyBuilder) ForCommodityAndAttribute(commType proto.CommodityDTO_CommodityType,
+	changedAttr proto.ActionItemDTO_CommodityAttribute) *HorizontalScaleMergePolicyBuilder {
+	comm := &CommodityMergeData{
+		commType:    commType,
+		changedAttr: changedAttr,
+	}
+	hsb.commTypes = append(hsb.commTypes, comm)
+	return hsb
+}
+
+// Create the ActionMergePolicyDTO for Horizontal Scale actions.
+func (hsb *HorizontalScaleMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error) {
+	if hsb.entityType == nil {
+		return nil, fmt.Errorf("Entity type required for horizontal scale merge policy")
+	}
+
+	if len(hsb.aggregationTargets) == 0 && len(hsb.chainedAggregationTargets) == 0 {
```

Comment:
Are you sure we need the `aggregationTargets` and that they should be required?

Reply:
removed as commented above. Thanks!